server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -738,12 +755,17 @@ public Map<String, Object> getIngestMetadata() {
     }
 
     /**
-     * Returns the document including its metadata fields, unless {@link #extractMetadata()} has been called, in which case the
-     * metadata fields will not be present anymore.
-     * Modify the document instead using {@link #setFieldValue(String, Object)} and {@link #removeField(String)}
+     * Fetch the timestamp from the ingestMetadata, if it exists
+     * @return the timestamp for the document or null
      */
-    public Map<String, Object> getSourceAndMetadata() {
-        return this.sourceAndMetadata;
+    public static ZonedDateTime getTimestamp(Map<String, Object> ingestMetadata) {
```

Comment:
This seems like an odd place for a utility method. Is there another place it could live, perhaps on metadata itself?

Reply:
Moved to IngestSourceAndMetadata.